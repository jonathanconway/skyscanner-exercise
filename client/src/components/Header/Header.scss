@import '~bpk-mixins';

@import '../../theme';

.Header {
  display: flex;
  padding: $bpk-spacing-xs;
  justify-content: space-between;
  align-items: center;
  background-color: $bpk-color-white;

  @include bpk-box-shadow-sm;

  &__hidden-text {
    @include bpk-visually-hidden;
  }

  &__logo-image {
    height: $bpk-spacing-base;
    vertical-align: top;
  }

  &__menu {
    fill: $brand-color;
  }
}
