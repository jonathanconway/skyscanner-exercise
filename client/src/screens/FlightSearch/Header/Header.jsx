import React, { useContext } from 'react';
import BpkText from 'bpk-component-text';
import BpkSmallLongArrowRight from 'bpk-component-icon/sm/long-arrow-right';

import FlightSearchContext from '../FlightSearch.context';

import STYLES from './Header.scss';

const c = className => STYLES[className] || 'UNKNOWN';

const Header = () => {
  const {
    originAirportCode,
    destinationAirportCode,
    numberOfTravellers,
    flightClass
  } = useContext(FlightSearchContext);

  return (
    <div className={c('Header')}>
      <BpkText tagName="h1" textStyle="xxl" className={c('Header__title')}>
        <span>{originAirportCode}</span>
        <BpkSmallLongArrowRight className={c('Header__title-arrow')} />
        <span>{destinationAirportCode}</span>
      </BpkText>
      <BpkText tagName="p" textStyle="base" className={c('Header__subtitle')}>
        {numberOfTravellers} travellers, {flightClass}
      </BpkText>
    </div>
  );
}

export default Header;