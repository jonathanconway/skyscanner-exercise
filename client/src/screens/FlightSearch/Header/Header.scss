@import '~bpk-mixins';

@import '../../../theme';

.Header {
  display: flex;
  flex-direction: column;
  background-color: $brand-color;
  color: $bpk-color-white;

  &__title {
    display: flex;
    margin: 0 0 $bpk-spacing-xs 0;
    line-height: $bpk-line-height-xl;

    & > * {
      align-self: center;
    }

    &-arrow {
      fill: $bpk-color-white;
      margin: 0 $bpk-spacing-xs;
    }
  }

  &__subtitle {
    margin: 0;
    line-height: $bpk-line-height-sm;
  }
}