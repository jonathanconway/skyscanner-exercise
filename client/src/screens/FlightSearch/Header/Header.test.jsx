import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';

import Header from './Header';

describe('Header', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<Header />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Header />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});