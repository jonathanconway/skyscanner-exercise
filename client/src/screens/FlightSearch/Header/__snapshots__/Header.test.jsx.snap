// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly 1`] = `
<div
  className="Header"
>
  <BpkText
    bold={false}
    className="Header__title"
    tagName="h1"
    textStyle="xxl"
  >
    <span />
    <_default
      className="Header__title-arrow"
    />
    <span />
  </BpkText>
  <BpkText
    bold={false}
    className="Header__subtitle"
    tagName="p"
    textStyle="base"
  >
     travellers, 
  </BpkText>
</div>
`;
